docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+// returns - did: Generated DID
+pub fn indy_create_did(did_resolver_handle,
+                       method,
+                       options) -> Future<did>
+```
+
+#### Create a new key pair associated with DID
+
+```Rust
+// Create a new key pair associated with DID.
+//
+// It verifies that key options are compatible with method,
+// creates and stores key in the wallet, creates and stores association
+// of this key with did.
+//
+// Note: It can produce the same data as indy_create_my_did.
+//
+// did_resolver_handle: resolver handle
+// did: did to associate with this key
+// options: some options like key type and seed
+// returns - key_json: json that describes the key. Can be passed to did
+//                     doc builder
+pub fn indy_create_did_key(did_resolver_handle,
+                           did,
+                           options) -> Future<key_json>
+```
+
+### DID Doc creation, updating and publishing
+
+#### Create a new DID Doc builder instance
+
+```Rust
+// Create a new DID Doc builder instance.
+//
+// It tries to resolve did doc for provided did and creates builder that
+// allows to modify did doc and publish it. If did doc isn't exists it
+// starts building from scratch
+//
+// did_resolver_handle: resolver handle
+// did: did to associate did doc
+// options: some options like key type and seed
+// returns - ddoc_builder_handle: did doc builder handle
+pub fn indy_create_did_doc_builder(did_resolver_handle,
+                                   did,
+                                   options) -> Future<ddoc_builder_handle>
+```
+
+#### Add new public key item to DID Doc builder
+
+```Rust
+// Add new public key item to DID Doc builder.
+//
+// Note: id can be full DID Doc reference or fragment.
+// In case of fragment reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// key_json: key_json in format compatible with did method
+// returns - ():
+pub fn indy_did_doc_add_public_key(ddoc_builder_handle,
+                                   id,
+                                   key_json) -> Future<()>
+```
+
+#### Update public key item in DID Doc builder
+
+```Rust
+// Update public key item in DID Doc builder.
+//
+// Note: id can be full DID Doc reference or fragment.
+// In case of fragment reference will be related to current DID.
+//
+// ddoc_builder_handle: builder handle
+// id: identifier of key
+// key_json: key_json in format compatible with did method
+// returns - ():
+pub fn indy_did_doc_update_public_key(ddoc_builder_handle,
+                                      id,
+                                      key_json) -> Future<()>
+```
+
+#### Delete public key item in DID Doc builder
+
+```Rust
+// Delete public key item in DID Doc builder.
+//
+// Note: id can be full DID Doc reference or fragment.
+// In case of fragment reference will be related to current DID.
+//
+// Note: If key is referenced inside of DID Doc validation error will
```

Comment:
What is meant by "referenced inside of DID Doc"?

Reply:
> What is meant by "referenced inside of DID Doc"?

It means used in some DID Doc sections. For example, in authentication section.